### ISSUE TITLE
Segfault Icinga2

### ISSUE THREAD

**Issue by harisspa** [Icinga/icinga2#8037]
Icinga2 version - 2.11.3-1

May 24 09:31:21 master-icinga-1a kernel: icinga2[531]: segfault at 48 ip 000000000062a617 sp 00007fbd824b2990 error 4 in icinga2[400000+a0c000]

We are stuck with this error for almost a week now. Don't see any errors when restarting Icinga2. Only when we check the status of Icinga2 and other logs we see this.

## To Reproduce

We are unable to start Icinga2 itself, and we are not sure on how to reproduce

## Expected behavior

Icinga2 restarts and reloads successfully and allows us getting alerts and notifications. The setup was working fine for almost 2, 3 months until it crashed this way

* Version used : version: 2.11.3-1
* Operating System and version: CentOS Linux 7 (Core)
* Enabled features (`icinga2 feature list`): 
                    Disabled features: command compatlog elasticsearch gelf influxdb opentsdb perfdata
                     Enabled features: api checker debuglog graphite ido-pgsql livestatus mainlog notification statusdata syslog

* Icinga Web 2 version and modules (System - About):
                  Icinga Web 2 Version2.7.3Git commit06cabfe8ba28cf545a42c92f25484383191a4e51PHP 
                  Version7.1.30Git commit date2019-10-18
* Config validation (`icinga2 daemon -C`): [2020-05-28 05:51:16 +0000] information/cli: Icinga application loader (version: 2.11.3-1)
[2020-05-28 05:51:16 +0000] information/cli: Loading configuration file(s).
[2020-05-28 05:51:17 +0000] information/ConfigItem: Committing config item(s).
[2020-05-28 05:51:17 +0000] information/ApiListener: My API identity: master-icinga-1a
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1 SyslogLogger.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1 LivestatusListener.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 31 HostGroups.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 2 FileLoggers.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 2 NotificationCommands.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1 NotificationComponent.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1657 Notifications.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1 IcingaApplication.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 176 Hosts.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1 ApiListener.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1 GraphiteWriter.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1 CheckerComponent.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 179 Zones.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1 StatusDataWriter.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 177 Endpoints.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 2 ApiUsers.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 239 CheckCommands.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1 IdoPgsqlConnection.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 3 TimePeriods.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1 UserGroup.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1 User.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 1481 Services.
[2020-05-28 05:51:17 +0000] information/ConfigItem: Instantiated 5 ServiceGroups.
[2020-05-28 05:51:17 +0000] information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
[2020-05-28 05:51:17 +0000] information/cli: Finished validating the configuration file(s).

* If you run multiple Icinga 2 instances, the `zones.conf` file (or `icinga2 object list --type Endpoint` and `icinga2 object list --type Zone`) from all affected nodes.

**Comment by lippserd** [comment]
Hi,

Thanks for the report. This may be a duplicate of #7754.

We're building test packages of the upcoming 2.11.4 release for you right now. I get back to you when the packages are ready.

All the best,
Eric

**Comment by lippserd** [comment]
Could you please test the following packages: https://git.icinga.com/packaging/rpm-icinga2/-/jobs/52339

Also, please make sure to enable core dumps: https://icinga.com/docs/icinga2/latest/doc/21-development/#core-dump.

**Comment by harisspa** [comment]
@lippserd  thanks. Will check and update you in sometime

**Comment by harisspa** [comment]
Hi @lippserd ,

error: Failed dependencies:
        icinga2-bin = 2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga is needed by icinga2-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64
        icinga2-common < 2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga conflicts with icinga2-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64
 file /usr/lib64/icinga2/sbin/icinga2 from install of icinga2-bin-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64 conflicts with file from package icinga2-bin-2.11.3-1.el7.icinga.x86_64
        file /usr/lib64/nagios/plugins/check_nscp_api from install of icinga2-bin-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64 conflicts with file from package icinga2-bin-2.11.3-1.el7.icinga.x86_64
        file /usr/lib/debug/usr/lib64/icinga2/libmysql_shim.so.2.11.3.debug from install of icinga2-debuginfo-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64 conflicts with file from package icinga2-debuginfo-2.11.3-1.el7.icinga.x86_64
        file /usr/lib/debug/usr/lib64/icinga2/libpgsql_shim.so.2.11.3.debug from install of icinga2-debuginfo-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64 conflicts with file from package icinga2-debuginfo-2.11.3-1.el7.icinga.x86_64
        file /usr/lib/debug/usr/lib64/icinga2/sbin/icinga2.debug from install of icinga2-debuginfo-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64 conflicts with file from package icinga2-debuginfo-2.11.3-1.el7.icinga.x86_64
        file /usr/lib/debug/usr/lib64/nagios/plugins/check_nscp_api.debug from install of icinga2-debuginfo-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64 conflicts with file from package icinga2-debuginfo-2.11.3-1.el7.icinga.x86_64
error: Failed dependencies:
        icinga2-bin = 2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga is needed by icinga2-ido-mysql-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64
error: Failed dependencies:
        icinga2-bin = 2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga is needed by icinga2-ido-pgsql-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64
error: Failed dependencies:
        icinga2-bin = 2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga is needed by icinga2-selinux-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64

We are getting these errors when we tried to directly install the RPMs.

**Comment by lippserd** [comment]
You have to mark all RPMs for the update, e.g. `ls icinga2*.rpm | grep -v pgsql | xargs rpm -Uvh` which installs all icinga2 RPMs except pgsql.

**Comment by harisspa** [comment]
3 Packages were installed successfully already. The errors pasted earlier are of the failed ones. 

package icinga2-common-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64 is already installed
package icinga2-doc-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64 is already installed
package icinga2-debuginfo-2.11.3.2.g13df2fc-0.20200528.0724.el7.icinga.x86_64 is already installed

The issue still persists even after upgrading these 3 packages.

**Comment by lippserd** [comment]
Could you please upload the core dump(s) [here](https://nextcloud.icinga.com/index.php/s/L74jFj5jcdeqn6x)?

**Comment by harisspa** [comment]
Icinga2 didn't start up at all, no crash dumps and core dumps were generated. Unable to generate manually as no PID available as the icinga2 process itself is not running. We were unable to upgrade other failed packages as well.

How long would it take to get the complete new release, so that we can try a proper upgrade to the existing installation.

**Comment by lippserd** [comment]
Without core dumps we can't guarantee that this issue is fixed with version 2.11.4.
You may use `yum reinstall` or remove and reinstall Icinga 2 from the RPMs.
We use those packages from our build system quite a lot and never had problems installing them.
You may start Icinga 2 in foreground with `icinga2 daemon -x debug` in order to see why the startup fails.

**Comment by harisspa** [comment]
I ran icinga2 daemon -x debug, didn't get any errors or warnings. However went through each conf and other files icinga2 going through. I noticed that 
[2020-05-28 16:55:43 +0000] information/ConfigObject: Restoring program state from file '/var/lib/icinga2/icinga2.state'

The file was last updated on May 21st 2020, and Icinga2 is still trying to reference that. I emptied the file, restarted Icinga2 and it has started successfully. Let me know of your views on this fix.

Will leave Icinga2 to run for a day and if we don't encounter any problem we are good to close this bug report from my side.
